client/app/components/Link.tsx

```diff
@@ -1,31 +1,57 @@
 import React from "react";
 import Button, { ButtonProps as AntdButtonProps } from "antd/lib/button";
 
-interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "role"> {
-  href: string;
+function DefaultLinkComponent({ children, ...props }: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
+  return <a {...props}>{children}</a>;
 }
 
-interface ButtonProps extends AntdButtonProps {
+Link.Component = DefaultLinkComponent;
+
+interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "role" | "type" | "target"> {
   href: string;
 }
+function Link({ children, ...props }: LinkProps) {
+  return <Link.Component {...props}>{children}</Link.Component>;
+}
 
-function DefaultLinkComponent({ children, ...props }: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
-  return <a {...props}>{children}</a>;
+interface LinkWithIconProps extends LinkProps {
+  children: string;
+  icon: JSX.Element;
+  alt: string;
+  target?: "_blank" | "_parent" | "_top";
```

Comment:
```suggestion
  target?: "_self" | "_blank" | "_parent" | "_top";
```

Reply:
removed `_self` so that just omitting target is preferred over it. If I allow it, the external link will be allowed to use `target="_self"`, which is a behavior intended to be reserved to `<Link.WithIcon>`. WDYT?

Reply:
Makes sense 👍